### DISCUSSION TITLE
ch05/external_jq.py goes wrong

### DISCUSSION THREAD

**Discussion by retsyo** [rtp-book/code#7] [Q&A] [answered]
I can run ch02-ch04 without problem with my python 3.9

however, fro ch05, when the HTML has been loaded, I get something wrong
Any hints? Thanks

![bug](https://github.com/rtp-book/code/assets/7960913/ffc8cff8-343b-4030-abbf-4343415a33f9)

**Comment by JennaSys** [comment, marked as answer]
It looks like jquery is not loading properly when the html file loads.

Are you able to get to the jquery URL by surfing to it directly just in the web browser or do you get an error there too?
https://code.jquery.com/jquery-3.5.1.min.js

The URL above is from line 4 of the HTML file:
 https://github.com/rtp-book/code/blob/77a959cd2e0aa8056f926e76f8eb2aef37440ba2/ch05/external_jq.html#L4

**Reply by retsyo** [reply]
Maybe there is something wrong with my internet connection. Thing goes well after I reboot my PC. Thanks
Later I will try to learn how to host every JavaScript codes on my PC.

**Reply by JennaSys** [reply]
Later on in the book, it gets into installing the JavaScript libraries locally and using a build tool for creating a JS bundle that gets sent to the web browser, instead of relying on the dynamic loading libraries from the internet. That will also allow your application to work without the internet being available at run-time.
